src/client/Client.cpp

```diff
@@ -29,7 +30,14 @@ void Client::draw(sf::RenderTarget& target, sf::Shader* shader, bool use_shader)
     // Draw active client groups
     for (auto client_group : mClientGroups) {
         if (client_group->isActive()) {
-            client_group->getCircle()->draw(target, shader, use_shader);
+            bool groupable = client_group->getGroupable();
+            std::shared_ptr<Circle> circle = client_group->getCircle();
+            if (groupable) {
+                circle->setColor(sf::Color(255, 0, 0));
+            } else {
+                circle->setColor(sf::Color(0, 0, 255));
```

Comment:
It makes sense for us to change the color or apply a shader on the client based on `groupable` but the default color of circles should be stored/set internally on the *server* not the client. With your change here, it's impossible to have different color circles for each client.

Reply:
previously it was just assigning a random color at boot and it wasn't communicating the color to the server

i agree with you the server should keep the colors or do something clever using the client_id to determine the color.... (client_id indexed into a pre-shared color array?)

but it's also a separate change, i dont think it's in-scope for this PR

Reply:
fixed in 92f3e38

Reply:
Color is not being synced from the server to the client